File: Easy/2026-04-10-3740-Minimum-Distance-Between-Three-Equal-Elements-I/solution.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        from collections import defaultdict
        indices = defaultdict(list)
        
        for i, num in enumerate(nums):
            indices[num].append(i)
        
        min_distance = float('inf')
        found = False
        
        for idx_list in indices.values():
            if len(idx_list) >= 3:
                for i in range(len(idx_list) - 2):
                    for j in range(i + 1, len(idx_list) - 1):
                        for k in range(j + 1, len(idx_list)):
                            distance = abs(idx_list[i] - idx_list[j]) + abs(idx_list[j] - idx_list[k]) + abs(idx_list[k] - idx_list[i])
                            min_distance = min(min_distance, distance)
                            found = True
        
        return min_distance if found else -1

File: Easy/2026-04-10-3740-Minimum-Distance-Between-Three-Equal-Elements-I/solution.js
var minimumDistance = function(nums) {
    const indices = new Map();
    let minDistance = Infinity;

    for (let i = 0; i < nums.length; i++) {
        if (!indices.has(nums[i])) {
            indices.set(nums[i], []);
        }
        indices.get(nums[i]).push(i);
    }

    for (const indexList of indices.values()) {
        if (indexList.length >= 3) {
            for (let i = 0; i < indexList.length - 2; i++) {
                for (let j = i + 1; j < indexList.length - 1; j++) {
                    for (let k = j + 1; k < indexList.length; k++) {
                        const distance = Math.abs(indexList[i] - indexList[j]) +
                                         Math.abs(indexList[j] - indexList[k]) +
                                         Math.abs(indexList[k] - indexList[i]);
                        minDistance = Math.min(minDistance, distance);
                    }
                }
            }
        }
    }

    return minDistance === Infinity ? -1 : minDistance;
};